.editorconfig

```diff
@@ -0,0 +1,28 @@
+root = true
+
+[*]
+end_of_line = lf
+charset = utf-8
+trim_trailing_whitespace = true
+
+[vcbuild.bat]
+end_of_line = crlf
+
+[*.{md,markdown}]
+trim_trailing_whitespace = false
+
+[{lib,src,test}/**.js]
+indent_style = space
+indent_size = 2
+
+[src/{**.h,**.cc}]
```

Comment:
I guess we could. C++ and JS are such different worlds though that I don't mind the verbosity of specifying them separately. That makes it clear that it was no accident, and that it's more incidental that both have the same 2 rules. The same can be said about test/*.py and configure (arguably, I could split up those as well at lines 22-24).

Your call :)